### PR TITLE
[MatrixCamp] add function cot

### PR DESCRIPTION
**Pull request by ZtXavier**
**What type of PR is this?**

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

**Which issue(s) this PR fixes:**

issue #1969

**What this PR does / why we need it:**

implement Cot builtin function

**Special notes for your reviewer:**

Not Available

**Additional documentation (e.g. design docs, usage docs, etc.):**

Not Available

### CODE REVIEW

File: pkg/builtin/unary/cot.go [hunk 1]
```diff
@@ -0,0 +1,221 @@
+package unary
```
Comment (broccoliSpicy) [inline pkg/builtin/unary/cot.go:1]: could you please add the license header...

File: pkg/vectorize/cot/cot_test.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
+package cot
```
Comment (broccoliSpicy) [inline pkg/vectorize/cot/cot_test.go:1]: the indentation format of this file doesn't consistent with the format above

**Review by broccoliSpicy — CHANGES_REQUESTED** [review]

File: pkg/builtin/unary/cot.go (hunk 1 above)
Comment (ZtXavier) [reply pkg/builtin/unary/cot.go:1]: ok

**Review by domingozhang — COMMENTED** [review]
bvt passed

**Comment by yingfeng** [comment]
Pls resolve the confliction before it's merged

**Comment by ZtXavier** [comment]
> 请在合并之前解决冲突

I have done

**Review by yingfeng — CHANGES_REQUESTED** [review]
Pls fix confliction caused by other merged PRs and make sure it could work with latest main

**Review by broccoliSpicy — CHANGES_REQUESTED** [review]

File: pkg/vectorize/cot/cot_test.go (hunk 2 above)
Comment (broccoliSpicy) [reply pkg/vectorize/cot/cot_test.go:14]: well, from my GitHub page, the indentation format doesn't seem right, I think you indented two spaces here in this file, but a tabulator in the other files above

**Review by broccoliSpicy — CHANGES_REQUESTED** [review]

**Comment by ZtXavier** [comment]
I have tried gofmt

**Review by broccoliSpicy — APPROVED** [review]
looks good to me

**Comment by ZtXavier** [comment]
> looks good to me
the first time to do this,thank you for your help

**Comment by broccoliSpicy** [comment]
> looks good to me
> the first time to do this,thank you for your help

haha, don't worry about it, I make all kinds of dump mistakes all the time

File: pkg/vectorize/cot/cot.go [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+	return rs
+}
+
+func CotUint16(xs []uint16, rs []float64) []float64 {
+	return cotUint16(xs, rs)
+}
+
+func CotUint16Pure(xs []uint16, rs []float64) []float64 {
+	for i, n := range xs {
+		rs[i] = 1.0 - math.Tan(float64(n))
```
Comment (broccoliSpicy) [inline pkg/vectorize/cot/cot.go:63]: cot --> 1.0 / tan

**Pull request merged**